tests/FixerFactoryTest.php

```diff
@@ -250,6 +251,82 @@ public function testNoDoubleConflictReporting()
         );
     }
 
+    public function testSetWhitespacesConfig()
+    {
+        $factory = new FixerFactory();
+        $config = new WhitespacesFixerConfig();
+
+        $fixer = $this->prophesize('PhpCsFixer\Fixer\WhitespacesAwareFixerInterface');
+        $fixer->getName()->willReturn('foo');
+        $fixer->setWhitespacesConfig($config)->shouldBeCalled();
+
+        $factory->registerFixer($fixer->reveal(), false);
+
+        $factory->setWhitespacesConfig($config);
+    }
+
+    public function testRegisterFixerInvalidName()
+    {
+        $this->setExpectedException('UnexpectedValueException', 'Fixer named "0" has invalid name.');
+
+        $factory = new FixerFactory();
+
+        $fixer = $this->createFixerDouble('0');
+
+        $factory->registerFixer($fixer, false);
+    }
+
+    public function testConfigureNonConfigurableFixer()
+    {
+        $this->setExpectedException(
+            'PhpCsFixer\ConfigurationException\InvalidFixerConfigurationException',
+            '[non_configurable] Is not configurable.'
+        );
+
+        $factory = new FixerFactory();
+
+        $fixer = $this->createFixerDouble('non_configurable');
+        $factory->registerFixer($fixer, false);
+
+        $factory->useRuleSet(new RuleSet(array(
+            'non_configurable' => array('bar' => 'baz'),
+        )));
+    }
+
+    public function testConfigureFixerWithNonArray()
+    {
+        $this->setExpectedException(
+            'PhpCsFixer\ConfigurationException\InvalidFixerConfigurationException',
+            '[foo] Configuration must be an array and may not be empty.'
+        );
+
+        $factory = new FixerFactory();
+
+        $fixer = $this->prophesize('PhpCsFixer\Fixer\ConfigurableFixerInterface');
+        $fixer->getName()->willReturn('foo');
+
+        $factory->registerFixer($fixer->reveal(), false);
+
+        $factory->useRuleSet(new RuleSet(array(
+            'foo' => 'bar',
```

Comment:
You could use a data provider to test more values (e.g. `null`, `array()`, ...).

Reply:
I can't test anything falsey here as it gets filtered out here https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/2.2/src/RuleSet.php#L294, not sure if that's the intended behaviour or not

Reply:
I can't say whether this is intended either, @SpacePossum @keradus do you have an idea?

Reply:
if you comment out that line maybe a test will tell, but I think it is by design to remove rules set to `false`, for example to remove a rule in the result set that got it in through inheritance but set to false by child. That besides `false` others like `null` are filtered as well might be an implementation detail though.

Reply:
that is exactly what the comment above that line states for:
>  filter out all resolvedRules that are off

only rules configured to `true | array` are supposed to survive

Reply:
An empty array also gets filtered out here

Reply:
empty array is invalid configuration

Reply:
Those values are invalid but currently most are silently ignored instead of throwing an exception. I wonder if this behavior is intended.